include/sound/sof.h

```diff
@@ -89,6 +89,7 @@ struct snd_sof_pdata {
 	/* machine */
 	struct platform_device *pdev_mach;
 	const struct snd_soc_acpi_mach *machine;
+	const void *mach_info;
```

Comment:
@dbaluta out of curiosity how do you handle for the IMX platforms?

Reply:
@ranj063, in our internal tree we read this info from device tree.

https://www.spinics.net/lists/devicetree/msg433908.html

We got lots of advice from community on how to handle this, but after several iterations i got busy with other stuff. I think @chunxu-li is the way to go. Will update with similar patches for i.MX